modules/infra/cli/notifications.c

```diff
@@ -130,6 +130,40 @@ static cmd_status_t notifications_dump(const struct gr_api_client *c, const stru
 				       &r6->nh);
 			}
 			break;
+		case NEXTHOP_EVENT_NEW:
+			if (n->payload_len == sizeof(*nh)) {
+				nh = (struct gr_nexthop *)&n[1];
+				if (nh->family == AF_INET)
+					printf("NH new: iface %d vrf %d %4p " ETH_F "\n",
+					       nh->iface_id,
+					       nh->vrf_id,
+					       &nh->ipv4,
+					       &nh->mac);
+				else if (nh->family == AF_INET6)
+					printf("NH new: iface %d vrf %d %6p " ETH_F "\n",
+					       nh->iface_id,
+					       nh->vrf_id,
+					       &nh->ipv6,
+					       &nh->mac);
```

Comment:
You can probably have a generic printf call here based on the family.

See infra/control/nexthop.c for reference:

https://github.com/DPDK/grout/blob/86f0506b731022ee17a2a255c1375add6afdaaf3/modules/infra/control/nexthop.c#L226-L229

That means you need to add a generic `addr` item in the address union in gr_nexthop as well:

https://github.com/DPDK/grout/blob/86f0506b731022ee17a2a255c1375add6afdaaf3/modules/infra/api/gr_nexthop.h#L28-L31

like so:

https://github.com/DPDK/grout/blob/86f0506b731022ee17a2a255c1375add6afdaaf3/modules/infra/control/gr_nh_control.h#L66-L71